### PR TITLE
KAFKA-5599: ConsoleConsumer : --new-consumer option as deprecated

### PR DESCRIPTION
**Pull request by ppatierno**
Marking --new-consumer as deprecated in the help of ConsoleConsumer
Printing a warning on using the new consumer but adding the --new-consumer option

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6099/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6114/
Test PASSed (JDK 7 and Scala 2.11).

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: Maybe at the end, "This is the default, so this option is deprecated and will be removed in a future release." Otherwise it may seem like using the new consumer is deprecated.

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ConsoleConsumer.scala]: Maybe:

"The --new-consumer option is deprecated and will be removed in a future major release. The new consumer is used by default if the --bootstrap-server option is provided."

**Review by ijuma — APPROVED** [review]
Thanks for the PR, left a couple of comments. Also, should we do this for all tools that have a `--new-consumer` option?

**Comment by ppatierno** [comment]
@ijuma updted after your comments. As native english speaker I think you are absolutely right ;) For the other tools which use the --new-consumer option I'm going to open a new JIRA and the providing a different PR.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6199/
Test FAILed (JDK 7 and Scala 2.11).

**Comment by ppatierno** [comment]
Please retest this

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6183/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by ppatierno** [comment]
Retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6195/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6211/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by ppatierno** [comment]
@ijuma any news about having this merged after my changes ?

**Comment by ppatierno** [comment]
This PR should be ready to merge after the @ijuma approval. What do you think @guozhangwang ? I have a quite similar PR #3555 for all other tools. It could be good to have both merged for the 1.0.0 release so we can then move to remove this option in the next release cycle.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6462/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6447/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by guozhangwang** [comment]
Thanks @ppatierno . I'll leave to @ijuma to merge then since he reviewed the PR.

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ConsoleConsumer.scala:272]: We should still have the description of the parameter. I'll add it back before merging.

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, LGTM. I'll do a minor change before merging to trunk as per the comment below.

**Review by ijuma — APPROVED** [review]

**Pull request closed without merging**
